Tests/SonarScanner.MSBuild.Tasks.UnitTests/IsTestFileByNameTests.cs

```diff
@@ -83,28 +83,27 @@ public void IsTestFile_NoRegex()
         public void IsTestFile_DefaultRegex()
         {
             var testFolder = TestUtils.CreateTestSpecificFolder(TestContext);
-            EnsureAnalysisConfig(testFolder, @"[^\\]*test[^\\]*$");
-
-            // filename contains 'test'
-            CheckFilePathIsTest(testFolder, "c:\\foo\\mytest.proj");
-            CheckFilePathIsTest(testFolder, "c:\\foo\\xtesty.proj");
-            CheckFilePathIsTest(testFolder, "c:\\foo\\bar space\\testmy.proj");
-            CheckFilePathIsTest(testFolder, "c:\\foo\\test.proj");
-            CheckFilePathIsTest(testFolder, "c:\\foo\\bar space\\foo.test");
-            CheckFilePathIsTest(testFolder, "c:\\foo\\bar space\\tEsT");
-            CheckFilePathIsTest(testFolder, "c:\\foo\\bar space\\xTestyyy.proj");
-            CheckFilePathIsTest(testFolder, "c:\\foo\\testtest");
-            CheckFilePathIsTest(testFolder, "c:\\foo\\bar\\a.test.proj");
-            CheckFilePathIsTest(testFolder, "c:\\foo\\test\\xtesty.proj");
-            CheckFilePathIsTest(testFolder, "xTESTy");
-            CheckFilePathIsTest(testFolder, "test");
-            CheckFilePathIsTest(testFolder, "c:\\foo\\ TEST ");
-
-            CheckFilePathIsNotTest(testFolder, "c:\\foo\\te st.proj");
-            CheckFilePathIsNotTest(testFolder, "c:\\foo\\bar\\test\\foo.proj");
-            CheckFilePathIsNotTest(testFolder, "c:\\test\\xtestyy\\foo");
-            CheckFilePathIsNotTest(testFolder, "c:\\foo\\bar\\myproj.csproj");
-            CheckFilePathIsNotTest(testFolder, "test\\foo.proj");
+            EnsureAnalysisConfig(testFolder, null);
+
+            CheckFilePathIsTest(testFolder, @"C:\Foo\MyProject.Test.csproj");
+            CheckFilePathIsTest(testFolder, @"C:\Foo\MyProject.Test.vbproj");
+            CheckFilePathIsTest(testFolder, @"C:\Foo\MyProject.Tests.csproj");
+            CheckFilePathIsTest(testFolder, @"C:\Foo\MyProject.Tests.vbproj");
+            CheckFilePathIsTest(testFolder, @"C:\Foo\MyProject.UnitTest.csproj");
+            CheckFilePathIsTest(testFolder, @"C:\Foo\MyProject.UnitTest.vbproj");
+            CheckFilePathIsTest(testFolder, @"C:\Foo\MyProject.UnitTests.csproj");
+            CheckFilePathIsTest(testFolder, @"C:\Foo\MyProject.UnitTests.vbproj");
+
+            // Doesn't end with regex
+            CheckFilePathIsNotTest(testFolder, @"C:\Foo\TestMyProject.csproj");
+            CheckFilePathIsNotTest(testFolder, @"C:\Foo\TestMyProject.vbproj");
+
+            // Case sensitive
+            CheckFilePathIsNotTest(testFolder, @"C:\Foo\MyProject.Unittest.csproj");
+            CheckFilePathIsNotTest(testFolder, @"C:\Foo\MyProject.Unittest.vbproj");
+
+            // Not expected extension
+            CheckFilePathIsNotTest(testFolder, @"C:\Foo\MyProject.Unittest.proj");
```

Comment:
Additional test cases:
* minimal matches e.g. Test.csproj, Tests.vbproj
* e.g. "Cutest.csproj" - something that shows the type of name we don't want to catch
* Test\Tests in the folder name -> not caught